spec/controllers/albums_controller_spec.rb

```diff
@@ -0,0 +1,110 @@
+require 'rails_helper'
+
+RSpec.describe AlbumsController, type: :controller do
+
+  let(:create_params) do {
+		album: {
+			name: "Harry Potter",
+			description: "Magic",
+      rank: 0
+		}
+	}
+  end
+	let(:album) { Album.create(create_params[:album]) }
+
+  describe "GET index" do
+    it "is successful" do
+      get :index
+      expect(response.status).to eq 200
+    end
+  end
+
+  describe "GET new" do
+    it "is successful" do
+      get :new
+      expect(response.status).to eq 200
+    end
+  end
+
+  describe "GET show" do
+    let(:album) do
+      Album.create(name: "Brawn")
+    end
+    it "renders the show view" do
+      get :show, id: album.id
+      expect(response.status).to eq 200
+    end
+  end
+
+  describe "GET edit" do
+    let(:album) do
+      Album.create(name: "Prawn")
+    end
+    it "renders the edit view" do
+      get :edit, id: album.id
+      expect(response.status).to eq 200
+    end
+  end
+
+  describe "POST 'create'" do
+
+    let(:bad_params) do
+      {
+        album: {
+          description: "the best"
+        }
+      }
+    end
+
+    it "redirects to show page" do
+      post :create, create_params
+      new_album = Album.last
+      expect(subject).to redirect_to album_path(new_album.id)
+    end
+
+    it "renders new template" do
```

Comment:
I like to make the description for clear -- "renders new template when create fails" and "redirects to show page when successful"